### PR TITLE
Add more security voters

### PR DESCRIPTION
**Pull request by leofeyer**
This applies the changes from #4648 to all similar cases.

### CODE REVIEW

**Review by aschempp — CHANGES_REQUESTED** [review]
Looks good so far! Two things I notice:

1. all the voters look very similar. I think we should have an abstract that only provides the supported table name and expects an action to vote on.
2. if we have full permissions implemented, we can also remove all the button callbacks (or actually the buttons!) in the DCA 🎉

**Comment by leofeyer** [comment]
Regarding 1: What about `AbstractListViewAccessVoter`, `AbstractParentViewAccessVoter` and `AbstractTreeViewAccessVoter`?

Regarding 2: Do I have to change anything in this PR?

**Comment by aschempp** [comment]
1: What's the difference betwee these abstract? To me the abstract should simple always return the same `supports` and have an `hasAccess` that is called in the `vote` method for the actual attribute. And the abstract would automatically make sure it only returns `ACCESS_DENIED`. I can do a PR on your branch if you'd like that.

2: I think these DCAs only have the default operations, so you can simply remove the `list.sorting.operations` completely and they will be auto-generated (same as in `tl_news_archive`).

**Comment by aschempp** [comment]
see https://github.com/leofeyer/contao/pull/7 for an example for calendars.

**Comment by aschempp** [comment]
can we get this into 5.3? 🙃

**Review by aschempp — COMMENTED** [review]
As discussed I have updated the PR and merged my abstract. Also added more voters for the child tables because their permissions are also pretty simple to test.

I wonder why the voters are all names `...AccessVoter`, the `Access` part does not make much sense to me. I would suggest to remove this everywhere, but for now I added it because it was already there. WDYT @contao/developers ?

File: core-bundle/src/Security/Voter/DataContainer/FavoritesVoter.php [hunk 1]
```diff
@@ -13,68 +13,48 @@
...
-    public function supportsType(string $subjectType): bool
+    protected function isGranted(CreateAction|DeleteAction|ReadAction|UpdateAction $action): bool
     {
-        return \in_array($subjectType, [CreateAction::class, ReadAction::class, UpdateAction::class, DeleteAction::class], true);
+        return match (true) {
+            $action instanceof CreateAction => true,
+            $action instanceof ReadAction,
+            $action instanceof UpdateAction,
+            $action instanceof DeleteAction => $this->checkAccess($action),
+            default => false,
```
Comment (aschempp) [inline core-bundle/src/Security/Voter/DataContainer/FavoritesVoter.php]: this should also be removed

File: news-bundle/src/Security/Voter/NewsArchiveAccessVoter.php [hunk 2]
```diff
@@ -12,54 +12,34 @@
...
-        }
-
-        return self::ACCESS_ABSTAIN;
+        return match (true) {
+            $action instanceof CreateAction => $this->security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES),
+            $action instanceof ReadAction,
+            $action instanceof UpdateAction => $this->security->isGranted(ContaoNewsPermissions::USER_CAN_EDIT_ARCHIVE, $action->getCurrentId()),
+            $action instanceof DeleteAction => $this->security->isGranted(ContaoNewsPermissions::USER_CAN_EDIT_ARCHIVE, $action->getCurrentId())
+                && $this->security->isGranted(ContaoNewsPermissions::USER_CAN_DELETE_ARCHIVES),
+            default => false,
```
Comment (aschempp) [inline news-bundle/src/Security/Voter/NewsArchiveAccessVoter.php]: this should also be removed

**Review by aschempp — CHANGES_REQUESTED** [review]

**Review by aschempp — APPROVED** [review]

**Pull request merged**